core/backend/src/CloudSqlite.ts

```diff
@@ -602,9 +602,224 @@ export namespace CloudSqlite {
+    private _cache?: CloudCache;
+    /** only for tests
+     * @internal
+     */
+    public setCache(cache: CloudCache) {
+      this._cache = cache;
+    }
+    /** @internal */
+    public getCache(): CloudCache {
+      return this._cache ??= this._ctor.getCacheForClass();
+    }
+    /** @internal */
+    public getCloudDb(): DbType {
+      return this._cloudDb;
+    }
+
+    /**
+     * The token that grants access to the cloud container for this DbAccess. If it does not grant write permissions, all
+     * write operations will fail. It should be refreshed (via a timer) before it expires.
+     */
+    public get sasToken() { return this._container.accessToken; }
+    public set sasToken(token: AccessToken) { this._container.accessToken = token; }
+
+    /** the container for this DbAccess. It is automatically connected to the CloudCache whenever it is accessed. */
+    public get container(): CloudContainer {
+      const container = this._container;
+      if (!container.isConnected)
+        container.connect(this.getCache());
+      return container;
+    }
+
+    /** Start a prefetch operation to download all the blocks for the VersionedSqliteDb */
+    public startPrefetch(): CloudPrefetch {
+      return startCloudPrefetch(this.container, this.dbName);
+    }
+
+    /** Create a new DbAccess for a database stored in a cloud container. */
+    public constructor(args: {
+      /** The Constructor for DbType. */
+      dbType: Constructor<DbType>;
+      /** The properties of the cloud container holding the database. */
+      props: ContainerAccessProps;
+      /** The name of the database within the container. */
+      dbName: string;
+    }) {
+      this._container = createCloudContainer({ ...args.props, writeable: true });
+      this._cloudDb = new args.dbType(args.props);
+      this.dbName = args.dbName;
+      this.lockParams.moniker = IModelHost.userMoniker;
+    }
+
+    /** Close the database for this DbAccess, if it is open */
+    public closeDb() {
+      if (this._cloudDb.isOpen)
+        this._cloudDb.closeDb();
+    }
+
+    /** Close the database for this DbAccess if it is opened, and disconnect this `DbAccess from its CloudContainer. */
+    public close() {
+      this.closeDb();
+      this._container.disconnect();
+    }
+
+    /**
+     * Initialize a cloud container to hold a Cloud SQliteDb. The container must first be created via its storage supplier api (e.g. Azure, or AWS).
+     * A valid sasToken that grants write access must be supplied. This function creates and uploads an empty database into the container.
+     * @note this deletes any existing content in the container.
+     */
+    protected static async _initializeDb(args: { dbType: typeof VersionedSqliteDb, props: ContainerAccessProps, dbName: string, blockSize?: number }) {
+      const container = createCloudContainer({ ...args.props, writeable: true });
+      container.initializeContainer({ blockSize: args.blockSize ?? 64 * 1024 });
```

Comment:
The comment on this function says 'the container must first be created via its storage supplier api', but initializeContainer will create the container if it doesn't exist I believe. Is that wording intentional to hopefully prevent people from using _initializeDb as a means of initializing the container?

Reply:
No, the comment is correct.  `initializeContainer` merely initializes a container for use by CloudSqlite, creating the manifest. It has no way to create a container, which must already exist and have an access token to write to it.

> Is that wording intentional to hopefully prevent people from using _initializeDb as a means of initializing the container

I don't  understand the question. This function *is* the way to initialize a container. It does mean that you can't use this function to store more than one database in a container though.

Reply:
initializeContainer calls sqlite3_bcv_create which has many comments in it implying that the container will be created if it does not exist, such as 

" First just try to upload the manifest. If that fails for any reason,
  ** assume the container does not exist and attempt to create it. If that
  ** succeeds, attempt to upload the manifest again. "

Reply:
That can only work with the storage emulator, with the account key. Nobody will have sufficient rights to create containers except the `BlobContainerService` - that's its purpose for existing!

Reply:
BTW, we should remove all mentions of account key from the entire CloudSqlite api, even for storage emulators. Account keys should never be used in any real code, even tests,

Reply:
Currently the BlobContainerService/cloudsqlitedbservice does have the CONNECTION_STRING which contains the account key. This value is part of the process's env.

Reply:
OK, i think @wgoehrig  told me the plan was to eliminate that.